src/scry/protocol/diagnostic.cr

```diff
@@ -1,46 +0,0 @@
-module Scry::Protocol
-  enum DiagnosticSeverity
-    Error       = 1
-    Warning     = 2
-    Information = 3
-    Hint        = 4
-  end
-
-  # Represents a diagnostic, such as a compiler error or warning.
-  # Diagnostic objects are only valid in the scope of a resource.
-  # See: https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#diagnostic
-  struct Diagnostic
-    JSON.mapping({
-      range:    Range,
-      severity: Int32,
-      source:   String,
-      message:  String,
-    }, true)
-
-    @filename = ""
-
-    # `uri` and `@filename` aren't part of Diagnostic interface of LSP,
-    # however They're useful to group failures by file.
-    def uri
-      if @filename.starts_with?("untitled:")
-        @filename
-      else
-        "file://#{@filename}"
-      end
-    end
-
-    def initialize(bf : BuildFailure)
```

Comment:
I'll write it here, even though it also applies to the other PR on `lsp` shard:

I think the lsp shard should be as language agnostic as possible, and this `BuildFailure` is specifically for the crystal compiler as far as I understand.

`BuildFailure` should remain in scry, and `LSP::Protocol::Diagnostic` should offer a constructor to initialize all its ivars, and remove any crystal-specific handling.

Do you see what I mean?

Reply:
Yep, that seems quite reasonable, thanks!